src/Keboola/DbExtractor/Application.php

```diff
@@ -4,73 +4,89 @@
 
 namespace Keboola\DbExtractor;
 
-use Keboola\Component\Logger;
+use Keboola\Component\BaseComponent;
 use Keboola\DbExtractor\Configuration\ActionConfigDefinition;
+use Keboola\DbExtractor\Configuration\BaseExtractorConfig;
 use Keboola\DbExtractor\Configuration\ConfigDefinition;
 use Keboola\DbExtractor\Configuration\ConfigRowDefinition;
-use Keboola\DbExtractor\Exception\UserException;
+use Keboola\Component\UserException;
 use Keboola\DbExtractor\Extractor\Extractor;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\Config\Definition\ConfigurationInterface;
 use Symfony\Component\Config\Definition\Exception\Exception as ConfigException;
-use Symfony\Component\Config\Definition\Processor;
-use ErrorException;
 
-class Application
+class Application extends BaseComponent
 {
-    /** @var string */
-    private $action;
-
-    /** @var array */
-    private $parameters;
-
-    /** @var array */
-    private $state;
-
-    /** @var Logger */
-    private $logger;
-
     /** @var Extractor */
     private $extractor;
 
     /** @var ConfigurationInterface */
     private $configDefinition;
 
-    public function __construct(array $config, Logger $logger, array $state = [])
-    {
-        static::setEnvironment();
+    /** @var string */
+    private $configDefinitionClassName;
+
+    public function __construct(
+        array $config,
+        LoggerInterface $logger,
+        array $state = [],
+        ?ConfigurationInterface $customConfigDefinition = null
+    ) {
```

Comment:
tohle ale znamena, ze to bude mit nestadnardni entrypoint (`run.php`), proc?

Reply:
To nechapu. Co znamena nestandardni entrypoint?

To jak se instancuje trida `Application` je v tom `run.php`. Nijak to neovlivnuje to, co leze do run.php

Reply:
jo a ten run.php by mel byt stejny jako v php-component

Reply:
To nechapu proc? Vpodstate cely `run.php` je stale stejny. Meni se jen to, co predas za parametry pri vytvareni instance "komponenty". Nevidim duvod proc by vpodstate kazda komponenta mela mit uplne stejny konstruktor, abychom docilili stejneho `run.php`. Nebo to nechapu a myslis neco jine?

Reply:
Ano, v ideálním případě bys to měl instancovat až v té Component. Component by měl být vlastně takový Controller. Prostě něco, co prováže nějak magicky infrastrukturu KBC s třídou, která dělá tu skutečnou práci.

Reply:
Jak pise tomas + nasledujici:

> Meni se jen to, co predas za parametry pri vytvareni instance "komponenty". 

a protoze zadny parametry nemas v run.php k dispozici, tak tam zadny nejsou (jen logger). Pokud by tam nejaky byly, tak to znamena, ze je nejak musis ziskat a timpadem vytahujes reusable kod z tridy `Component` do `run.php` kde to reusovatelny neni a musi se porad kopirovat.

Reply:
Sorry, neposlal jsem ti spravne info k tomu. Je to vyresene tady https://github.com/keboola/db-extractor-common/pull/94/commits/2a3ed64f1c03c031a11da66342492eaf2b5c4cb3

U tridy Application uz bych nic neresil, smazu ji v #94